src/common/query/src/prelude.rs

```diff
@@ -12,12 +12,46 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use common_base::regex_pattern::NAME_PATTERN_REG;
 pub use datafusion_common::ScalarValue;
+use once_cell::sync::OnceCell;
+use snafu::ensure;
 
 pub use crate::columnar_value::ColumnarValue;
+use crate::error::{DefaultTSColNameSnafu, Result};
 
-/// Default timestamp column name for Prometheus metrics.
-pub const GREPTIME_TIMESTAMP: &str = "greptime_timestamp";
+/// Default timestamp column name.
+static GREPTIME_TIMESTAMP_CELL: OnceCell<String> = OnceCell::new();
+
+/// Get the default timestamp column name.
+/// Returns the configured value, or `greptime_timestamp` if not set.
+pub fn greptime_timestamp() -> &'static str {
+    GREPTIME_TIMESTAMP_CELL.get_or_init(|| GREPTIME_TIMESTAMP.to_string())
+}
+
+/// Set the default timestamp column name.
+/// This should be called once during application startup.
+/// Returns Ok(()) if successful, or Err with the attempted value if already set.
+pub fn set_greptime_timestamp(name: Option<&str>) -> Result<()> {
+    let ts = match name {
+        None | Some("") => GREPTIME_TIMESTAMP,
+        Some(ts) => ts,
+    };
+
+    ensure!(
+        NAME_PATTERN_REG.is_match(ts),
+        DefaultTSColNameSnafu {
+            name: ts,
+            message: format!("Invalid character in timestamp column name: {}", ts)
+        }
+    );
+
+    GREPTIME_TIMESTAMP_CELL.get_or_init(|| ts.to_string());
+    Ok(())
+}
+
+/// Default timestamp column name constant for backward compatibility.
+const GREPTIME_TIMESTAMP: &str = "greptime_timestamp";
 /// Default value column name for Prometheus metrics.
 pub const GREPTIME_VALUE: &str = "greptime_value";
```

Comment:
Should `greptime_value` also be configurable?

Reply:
In that case, how about change the config option to something like `default_prefix`, so that it applies to time index column name and value column name. For example, set the option to `new_db`, and the column names would be `new_db_timestamp` and `new_db_value`

Reply:
Sounds good

Reply:
@killme2008 PTAL